gprofiler/main.py

```diff
@@ -604,13 +679,13 @@ def parse_cmd_args() -> configargparse.Namespace:
     args.perf_inject = args.nodejs_mode == "perf"
     args.perf_node_attach = args.nodejs_mode == "attach-maps"
 
-    if args.upload_results:
+    if args.upload_results or args.file_path:
```

Comment:
In one site you use `args.upload_results or args.subcommands == "upload-file"` and in this site you use `args.upload_results or args.file_path`. Let's be consistent. You can create a helper function that gets the `args` object and returns true/false if we should be uploading.

Alternatively - and possible simpler - is to add after the `parse_args` call:
```
if args.subcommand == "upload-file":
   args.upload_results = True
```
then we don't get the 2 branches of the code.